wasmcloud-wasi-kv/wit/keyvalue.wit

```diff
@@ -0,0 +1,50 @@
+// Due to the fact that muiltiple interfaces cannot be hung from one component
+// implementation (as of 2023/03/14), we must combine the keyvalue interface
+// and the wasmcloud actor interface into one.
+default interface wasmcloud-kv {
+	/// A keyvalue interface that provides simple read and write operations.
+	use pkg.types.{bucket, error, incoming-value, outgoing-value}
+
+	/// A key is a unique identifier for a value in a bucket. The key is used to
+	/// retrieve the value from the bucket.
+	type key = string
+
+	/// A list of keys
+	type keys = list<key>
+
+	/// Get the value associated with the key in the bucket. It returns a incoming-value
+	/// that can be consumed to get the value.
+	///
+	/// If the key does not exist in the bucket, it returns an error.
+	get: func(bucket: bucket, key: key) -> result<incoming-value, error>
+
+	/// Set the value associated with the key in the bucket. If the key already
+	/// exists in the bucket, it overwrites the value.
+	///
+	/// If the key does not exist in the bucket, it creates a new key-value pair.
+	/// If any other error occurs, it returns an error.
+	set: func(bucket: bucket, key: key, outgoing-value: outgoing-value) -> result<_, error>
+
+	/// Delete the key-value pair associated with the key in the bucket.
+	///
+	/// If the key does not exist in the bucket, it returns an error.
+	delete: func(bucket: bucket, key: key) -> result<_, error>
+
+	/// Check if the key exists in the bucket.
+	///
+	/// If the key does not exist in the bucket, it returns an error.
+	exists: func(bucket: bucket, key: key) -> result<bool, error>
+
+	/// Wasmcloud guest
+	guest-call: func(operation: string, payload: option<list<u8>>) -> result<option<list<u8>>, string>
```

Comment:
I'm confused, can't the actor just implement multiple interfaces instead?

Reply:
No, https://cosmonic.slack.com/archives/C04S539JRB8/p1678788310748789